app/services/random_topic_selector.rb

```diff
@@ -25,7 +25,7 @@ def self.backfill(category = nil)
       end
 
       # don't leak private categories into the "everything" group
-      options[:guardian] = Guardian.new(CategoryFeaturedTopic.fake_admin)
+      options[:guardian] = Guardian.new(Discourse.system_user)
```

Comment:
@SamSaffron, blame says you're the last to change this - can you recheck the security implications of this?